backend/src/test/java/com/github/bolabolor/backend/MongoUserDetailsServiceTest.java

```diff
@@ -0,0 +1,56 @@
+package com.github.bolabolor.backend;
+
+import com.github.bolabolor.backend.security.MongoUser;
+import com.github.bolabolor.backend.security.MongoUserRepository;
+import com.github.bolabolor.backend.security.MongoUsersDetailsService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+
+import java.util.Optional;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class MongoUserDetailsServiceTest {
+
+    private MongoUsersDetailsService detailsService;
+    @Mock
+    private MongoUserRepository mongoUserRepository;
+
+    @BeforeEach
+    void setup() {
+        detailsService = new MongoUsersDetailsService(mongoUserRepository);
+    }
+
+
+    @Test
+    void expectSuccessfulUsername() {
+        //Given
+        MongoUser mongoUser = new MongoUser(UUID.randomUUID().toString(),
+                "username", "password");
```

Comment:
Optional: Besser lesbar
```suggestion
        MongoUser mongoUser = new MongoUser(
                UUID.randomUUID().toString(),
                "username",
                "password");
```